src/cheri/rvy-cheriot-isa-unpriv.adoc

```diff
@@ -35,12 +34,8 @@ we have not yet found a compelling reason to formally specify this composition.
 
 [NOTE]
 =====
-While {cheriot_unpriv_ext_name} is nominally compatible with
-<<section_rvy_sentry_insn_ext>>,
-the operating system written for link:https://cheriot.org[CHERIoT]
-has a security model that presumes the absence of ambient sealing,
-and so this specification does not
-define any <<sec_cap_type_ambient,ambiently available>> sentry types.
+{cheriot_unpriv_ext_name} is incompatible with
+<<section_rvy_sentry_insn_ext>> as <<sec_cap_type>> of `1` is not a sentry type.
```

Comment:
Zycheriot (specifically, §14.1.2 "Capability Types" of the present riscv-cheri doc) _does_ designate otype `1` as a sentry type (which, again, the base defines as "JALR unseals this type"; see §2.3.9).  In fact, it's even _almost_ "the same" sentry as in the "default" encoding's implicit semantics (specifically, it's our forward-arc interrupt-posture-inheriting sentry type).  That said, there are some significant differences: CHERIoT does not use "HOOK 2" to have JALR _produce_ capabilities with type `1` (unlike the default's, where forward- and backward-arc types aren't distinguished) and CHERIoT's encodings cannot represent type `1` capabilities that do not grant X-permission.

Reply:
The solution here may well be to stop trying to make JALR generic and simply to have it in an extension with different versions for the different encoding formats.
If we can't work out how to make this work between ourselves we're not going to get it through review.
So maybe Zysentry has one vesion of JALR, and there's another for CHERIoT.
And CHERI implementations _must_ choose a version of JALR.
Then we can all write exactly what we want JALR to do without adding the hooks.

Would that solve it?

Reply:
I would just drop this diff from this PR. The current wording looks fine as is to me.